### PR TITLE
Executor: Add first NodeJS routines

### PR DESCRIPTION
**Pull request by sven-oly**

### CODE REVIEW

File: exectors/nodejs/collator.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+module.exports = {
+
+  testCollationShort: function(json) {
+
+    // Global default locale
+    let testLocale = '';
+    let testCollOptions = {ignorePunctuation:'true'};
+
+    // Set up collator object with optional locale and testOptions.
+    let coll;
```
Comment (gnrunge) [inline exectors/nodejs/collator.js:14]: As best practice: initialize with 0/null.

File: exectors/nodejs/displaynames.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// Tests Intl class DisplayNames
+
+module.exports = {
+
+  testDisplayNames: function (json) {
+    let locale = 'en';  // Default
+    let options;
```
Comment (gnrunge) [inline exectors/nodejs/displaynames.js:7]: Initialization!?

File: exectors/nodejs/numberformat.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
+// Tests NumberFormat and DecimalFormat classes
+
+  let patternsToOptions = new Map();
+  patternsToOptions.set("0.0", {minimumFractionDigits: 1});
+  patternsToOptions.set("00", {minimumSignificantDigits: 1});
+  patternsToOptions.set("0.0", {minimumFractionDigits: 1});
```
Comment (gnrunge) [inline exectors/nodejs/numberformat.js:6]: Duplicate of line 4. Either way, the mapping takes place in the data generator and this map can be deleted altogether.

File: exectors/nodejs/numberformat.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    const pattern = json['pattern'];
+    const rounding = json['rounding'];
+    const input = parseFloat(json['input']);
+
+    let options;
+    let error = "unimplemented pattern";
+
+    // If options are in the JSON, use them...
+    try {
```
Comment (gnrunge) [inline exectors/nodejs/numberformat.js:37]: Is there another way to find out if 'options' are available than try/catch? try/catch should be used mainly for error handling but not for testing for parameter availability.

File: exectors/nodejs/numberformat.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    if (!options) {
+      // Don't test, but return an error
+      return {"label": label,
+              "error": error,
+              "pattern": pattern,
+             };
+    }
+    let testLocale = json['locale'];
```
Comment (gnrunge) [inline exectors/nodejs/numberformat.js:55]: No try/catch here(?)

File: exectors/nodejs/executor.js [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+ */
+
+let doLogInput = false;
+
+// Test type support. Add new items as they are implemented
+const testTypes = {
+  TestCollShiftShort : Symbol("coll_shift_short"),
+  TestDecimalFormat : Symbol("decimal_fmt"),
+  TestNumberFormat : Symbol("number_fmt"),
+  TestDateTimeFormat : Symbol("datetime_fmtl"),
```
Comment (gnrunge) [inline exectors/nodejs/executor.js:48]: ..._fmtl. 'l'?

File: exectors/nodejs/numberformat.js (hunk 4 above)
Comment (sven-oly) [reply exectors/nodejs/numberformat.js:37]: Yes, fixed this.

File: exectors/nodejs/numberformat.js (hunk 3 above)
Comment (sven-oly) [reply exectors/nodejs/numberformat.js:6]: Removed the duplicate line. Yes, I'll remove this function when I get the options working.

File: exectors/nodejs/displaynames.js (hunk 2 above)
Comment (sven-oly) [reply exectors/nodejs/displaynames.js:7]: This is OK in JS because it sets to "undefined".

File: exectors/nodejs/collator.js (hunk 1 above)
Comment (sven-oly) [reply exectors/nodejs/collator.js:14]: This is OK in JS because it sets to "undefined".

**Pull request merged**